pipeline/extract.py

```diff
@@ -1,47 +1,45 @@
-"""Extracts/Scrapes the desired information from the required product."""
+"""
+Extracts/scrapes the desired information from the required product.
+"""
+import json
 from os import environ
-from urllib.request import urlopen
 
 from bs4 import BeautifulSoup
 from dotenv import load_dotenv
-import requests as req
+import requests
 
 
-def scrape_product_page(link: str, headers: dict) -> dict:
-    """Returns a dict of scraped details for a specific product."""
+def scrape_asos_page(url: str, header: dict) -> dict:
+    """
+    Scrapes an ASOS page and returns a dict of desired data about the product.
+    """
+    page = requests.get(url, headers=header)
+    soup = BeautifulSoup(page.text, "html.parser").find(
+        "script", type="application/ld+json")
+    product_data = json.loads(soup.string)
+    print(product_data)
 
-    res = req.get(link, headers=headers, timeout=5)
-    soup = BeautifulSoup(res.content, features="html.parser")
-    price = soup.find_all('price-screenreader-only-text')
-    print(price)
+    price_endpoint = f"https://www.asos.com/api/product/catalogue/v3/stockprice?productIds={product_data['productID']}&store=COM&currency=GBP"
 
-    product_price_element = soup.find(
-        'span', {'data-testid': 'price-screenreader-only-text'})
-    product = {
-        "name": soup.find("h1", class_="jcdpl").text.strip(),
-        "price":  product_price_element}
-
-    return product
+    price = requests.get(price_endpoint).json()[
+        0]["productPrice"]["current"]["text"]
 
```

Comment:
How will this handle instances where this is not found? Maybe a default would be useful.